key-manager-oapi.yaml

```diff
@@ -0,0 +1,223 @@
+
+        "401":
+          $ref: "#/components/responses/Unauthorized"
+        "403":
+          $ref: "#/components/responses/Forbidden"
+        "500":
+          $ref: "#/components/responses/InternalError"
+
+    post:
+      operationId: ImportKeystores
+      summary: Import Keystores.
+      description: |
+        Import keystores generated by the Eth2.0 deposit CLI tooling. All keystores MUST be encrypted with the same password.
+
+        Users SHOULD send slashing_protection data associated with the imported pubkeys. MUST follow the format defined in
+        EIP-3076: Slashing Protection Interchange Format.
+      security:
+        - cookieAuth: []
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                keystores:
+                  type: array
+                  description: JSON-encoded keystore files generated with the Launchpad.
+                  items:
+                    type: object
+                passwords:
+                  description: Passwords to unlock imported keystore files. `request.passwords[i]` must unlock `request.keystores[i]`.
+                  items:
+                    type: string
+                slashing_protection:
+                  type: object
+                  required: false
+                  description: Slashing protection of the imported keys data in EIP-3076 JSON format.
+      responses:
+        "200":
+          description: Success response
+          content:
+            application/json:
+              schema:
+                title: ImportKeystoresResponse
+                type: object
+                properties:
+                  statuses:
+                    type: array
+                    description: Status result of each `request.keystores` with same length and order of `request.keystores`
+                      type: object
+                      properties:
+                        status:
+                          type: string
+                          description: |
+                            - imported: Keystore successfully decrypted and imported to keymanager permanent storage
+                            - duplicate: Keystore's pubkey is already known to the keymanager
+                            - error: Any other status different to the above: decrypting error, I/O errors, etc.
+                          enum:
+                            - imported
+                            - duplicate
+                            - error
+                        message:
+                          type: string
+                          description: error message if status == error
+        "401":
+          $ref: "#/components/responses/Unauthorized"
+        "403":
+          $ref: "#/components/responses/Forbidden"
+        "500":
+          $ref: "#/components/responses/InternalError"
+
+    delete:
+      operationId: DeleteKeys
```

Comment:
question on intended behavior, if you send an empty request body is the intention to return the current state of slashing protection data? Also is the slashing protection data only on keys that are inactive or returning the data on the active ones as well 🤔

Reply:
probably never mind on the second part, it doesn't hurt to have more data right 🤔

Reply:
No, it's very risky to return slashing protection data for keys that haven't been deleted, as it represents stale information that could get you slashed.

Slashing protection data should only be returned for keys from the request that were successfully deleted (status == deleted). We should clarify that in the description